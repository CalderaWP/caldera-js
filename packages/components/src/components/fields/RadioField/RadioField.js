import PropTypes from 'prop-types';
import React from 'react';
import { RadioControl } from '@wordpress/components';
import { parseAttributes, fieldClassNames, isValidHtml5type } from '../util';
export const RadioField = ({
	attributes,
	label,
	fieldId,
	onChange,
	value,
	options,
	multiple,
	description,
	placeholder
}) => {
	attributes = parseAttributes(attributes, 'radio');

	return (
		<RadioControl
			className={fieldClassNames('radio')}
			id={fieldId}
			selected={value}
			options={options}
			onChange={onChange}
			label={label}
			help={description}
			{...attributes}
		/>
	);
};

RadioField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	required: PropTypes.bool,
	multiple: PropTypes.bool
};

RadioField.defaultProps = {
	required: false,
	multiple: false
};
