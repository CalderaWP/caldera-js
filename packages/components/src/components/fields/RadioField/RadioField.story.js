import React from 'react';
import { storiesOf } from '@storybook/react';
import { RadioField } from './RadioField';

const onChange = newValue => {
	console.log(newValue);
};
const onBlur = event => {};

const STORY_NAME_RADIO = 'RadioField';
storiesOf(STORY_NAME_RADIO, module).add('With no value selected ', () => (
	<RadioField
		label={'Select A Hat'}
		onChange={onChange}
		description={'selection of hats'}
		fieldId={'selection-hats'}
		required={true}
		options={[{ value: 100, label: 'One' }, { value: 200, label: 'Two' }]}
	/>
));
storiesOf(STORY_NAME_RADIO, module).add('With  value selected ', () => (
	<RadioField
		label={'Select A Hat'}
		onChange={onChange}
		description={'selection of hats'}
		fieldId={'selection-hats'}
		required={true}
		value={200}
		options={[{ value: 100, label: 'One' }, { value: 200, label: 'Two' }]}
	/>
));
