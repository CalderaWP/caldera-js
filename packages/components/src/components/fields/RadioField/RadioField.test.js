import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { RadioField } from './RadioField';

describe('RadioField Field component', () => {
	let onChange;
	beforeEach(() => {
		onChange = jest.fn();
	});
	it('matches snapshot', () => {
		const component = renderer.create(
			<RadioField
				label={'Select A Hat!'}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				options={[
					{ value: 100, label: 'Hat One' },
					{ value: 200, label: 'Hat Two' }
				]}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Fires on change event', () => {
		const component = mount(
			<RadioField
				label={'Select A Hat'}
				onChange={onChange}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				required={true}
				multiple={true}
				options={[
					{ value: 100, label: 'One' },
					{ value: 200, label: 'Two' }
				]}
			/>
		);

		const event = { target: { value: 200 } };
		component
			.find('input')
			.first()
			.simulate('change', event);
		expect(onChange.mock.calls.length).toBe(1);
		expect(onChange.mock.calls[0][0]).toBe(200);
	});
});
