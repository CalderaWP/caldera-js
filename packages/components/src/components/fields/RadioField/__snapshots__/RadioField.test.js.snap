// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioField Field component matches snapshot 1`] = `
<div
  className="components-base-control caldera-field caldera-field-radio components-radio-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-radio-control-0"
    >
      Select A Hat!
    </label>
    <div
      className="components-radio-control__option"
    >
      <input
        aria-describedby="inspector-radio-control-0__help"
        checked={false}
        className="components-radio-control__input"
        id="inspector-radio-control-0-0"
        name="inspector-radio-control-0"
        onChange={[Function]}
        type="radio"
        value={100}
      />
      <label
        htmlFor="inspector-radio-control-0-0"
      >
        Hat One
      </label>
    </div>
    <div
      className="components-radio-control__option"
    >
      <input
        aria-describedby="inspector-radio-control-0__help"
        checked={false}
        className="components-radio-control__input"
        id="inspector-radio-control-0-1"
        name="inspector-radio-control-0"
        onChange={[Function]}
        type="radio"
        value={200}
      />
      <label
        htmlFor="inspector-radio-control-0-1"
      >
        Hat Two
      </label>
    </div>
  </div>
  <p
    className="components-base-control__help"
    id="inspector-radio-control-0__help"
  >
    selection of hats
  </p>
</div>
`;
